--- mpd-0.19.17/src/decoder/plugins/WildmidiDecoderPlugin.cxx.orig	2016-07-17 21:33:47.160364664 +0100
+++ mpd-0.19.17/src/decoder/plugins/WildmidiDecoderPlugin.cxx	2016-07-17 21:35:09.376359407 +0100
@@ -101,7 +101,13 @@
 		if (info == nullptr)
 			break;
 
-		len = WildMidi_GetOutput(wm, buffer, sizeof(buffer));
+#if defined(LIBWILDMIDI_VERSION) && (LIBWILDMIDI_VERSION >= 0x000400)
+                len = WildMidi_GetOutput(wm, (int8_t *)buffer, sizeof(buffer));
+#else
+                len = WildMidi_GetOutput(wm, (char *)buffer, sizeof(buffer));
+#endif
+
+
 		if (len <= 0)
 			break;
 
